src/nft.vy

```diff
@@ -48,6 +48,8 @@
 interface IERC721Receiver:
     def onERC721Received(operator: address, _from: address, tokenId: uint256, data: Bytes[1024]) -> bytes4: nonpayable
 
+interface IAuctionContract:
+    def start_auction_with_auctionhouse_held_nft(lot: uint256, patron: address): nonpayable
```

Comment:
Suggestion (not a bug): What if we shift logic of minting to AuctionContract? 
Ex: `mint_and_start_auction`? 
- Keep NFT contract as simple as possible, no special logic for auction contract in NFT contract, NFT already have minter role functionality

Reply:
good point, and I prefer it too. implemented